quesma/optimize/shorten_time_range.go

```diff
@@ -0,0 +1,248 @@
+		return nil
+	}
+	if len(selectCommand.LimitBy) != 0 {
+		logger.Debug().Msg("Query not eligible for time range optimization: LIMIT BY")
+		return nil
+	}
+	if selectCommand.SampleLimit != 0 {
+		logger.Debug().Msg("Query not eligible for time range optimization: SAMPLE LIMIT")
+		return nil
+	}
+	if selectCommand.IsDistinct {
+		logger.Debug().Msg("Query not eligible for time range optimization: DISTINCT")
+		return nil
+	}
+	if selectCommand.GroupBy != nil {
+		logger.Debug().Msg("Query not eligible for time range optimization: GROUP BY")
+		return nil
+	}
+	if len(selectCommand.NamedCTEs) != 0 {
+		logger.Debug().Msg("Query not eligible for time range optimization: CTEs")
+		return nil
+	}
+
+	if !s.validateSelectedColumns(selectCommand.Columns) {
+		return nil
+	}
+
+	orderByColumnName, direction, found := s.findOrderByColumn(selectCommand)
+	if !found {
+		return nil
+	}
+
+	lowerLimit, upperLimit := s.findTimeLimits(selectCommand, orderByColumnName)
+	if lowerLimit == nil || upperLimit == nil {
+		logger.Debug().Msg("Query not eligible for time range optimization: missing time limits (both lower and upper)")
+		return nil
+	}
+
+	return &timeRange{columnName: orderByColumnName, lowerLimit: *lowerLimit, upperLimit: *upperLimit, direction: direction}
+}
+
+func (s shortenTimeRange) columnsToAliasedColumns(columns []model.Expr) []model.Expr {
+	aliasedColumns := make([]model.Expr, len(columns))
+	for i, column := range columns {
+		// Using aliasing rules from ProcessQuery
+
+		if columnRef, ok := column.(model.ColumnRef); ok {
+			aliasedColumns[i] = model.NewAliasedExpr(column, columnRef.ColumnName)
+			continue
+		}
+		if col, ok := column.(model.LiteralExpr); ok {
+			var colName string
+			if str, isStr := col.Value.(string); isStr {
+				if unquoted, err := strconv.Unquote(str); err == nil {
+					colName = unquoted
+				} else {
+					colName = str
+				}
+			} else {
+				if colName == "" {
+					colName = fmt.Sprintf("column_%d", i)
+				}
+			}
+			aliasedColumns[i] = model.NewAliasedExpr(column, colName)
+			continue
+		}
+		if aliasedExpr, ok := column.(model.AliasedExpr); ok {
+			aliasedColumns[i] = aliasedExpr
+			continue
+		}
+		if _, ok := column.(model.FunctionExpr); ok {
+			aliasedColumns[i] = model.NewAliasedExpr(column, fmt.Sprintf("column_%d", i))
+			continue
+		}
+
+		aliasedColumns[i] = model.NewAliasedExpr(column, fmt.Sprintf("column_%d", i))
+		logger.Error().Msgf("Quesma internal error - unreachable code: unsupported column type %T", column)
+	}
+	return aliasedColumns
+}
+
+func (s shortenTimeRange) transformQuery(query *model.Query) (*model.Query, error) {
+	foundTimeRange := s.findTimeRange(&query.SelectCommand)
+	if foundTimeRange == nil {
+		return query, nil
+	}
+
+	if foundTimeRange.direction != model.DescOrder {
+		// TODO: support other directions
+		return query, nil
+	}
+
+	splitPoint := foundTimeRange.upperLimit - 60*60*1000 // TODO: better split point
```

Comment:
Move to constant and possible to override by configuration.

Reply:
It's now configurable by YAML configuration:
```yaml
          optimizers:
            split_time_range:
              properties:
                ranges: "15,60,120"
```